---
title: "Targeted Nanopore sequencing for MGMT promoter methylation evaluation"
SUbtitle: "Figures and data transformation"
author: "Skarphedinn Halldorsson"
date: "Compiled on `r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document:
    latex_engine: pdflatex
    extra_dependencies: subfig, float
    number_sections: yes
    toc: no
  word_document:
    toc: no
    toc_depth: '3'
  html_document:
    toc: yes
    toc_depth: '3'
    df_print: paged
editor_options:
  chunk_output_type: console
subtitle: Preliminary data analysis
header-includes:
- \usepackage{fancyhdr}
- \usepackage{mathtools}
- \usepackage{hyperref}
- \usepackage{subfig}
- \setlength{\headheight}{33pt}
- \setlength{\footskip}{25pt}
- \pagestyle{fancy}
- \renewcommand{\headrulewidth}{0.5pt}
- \renewcommand{\footrulewidth}{0.5pt}
- \cfoot{\scriptsize Vilhelm Magnus Laboratory | Institute for surgical research \\
  Oslo University Hospital}
- \rhead{\thepage}
- \hypersetup{colorlinks   = true, linkcolor=blue, urlcolor  = blue}
- \fancypagestyle{plain}{\pagestyle{fancy}}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(dpi = 100,
                      echo=TRUE,
                      warning=FALSE, message=FALSE, eval = TRUE,
                      fig.show=TRUE, fig.width= 5,fig.height= 5,fig.align='center',
                      out.width = '100%')
```

## Figures


```{r Load data, echo=FALSE}

library(ggplot2)
library(dplyr)
library(tidyr)
library(janitor)
#library(kableExtra)
library(RColorBrewer)
library(scales)
library(tibble)

load("MGMT.Rdata")

# # select the relevant columns from the concatenated bedMethyl file including all samples 
# All_MGMT <- All_MGMT %>% select(2,10:19) %>% relocate(11)
# 
# # name the selected columns columns
# names(All_MGMT) <- c("SampleID", "Pos", "Valid_cov", "Methylation_percent", "N_mod", "N_canon", "N_other_mod", "N_delete", "N_fail", "N_diff", "N_nocall")

# Extract the 98 CpG sites in the MGMT CpG island
All_MGMT_Island <- All_MGMT %>% filter(between(Pos, 129466683, 129467448))

# Calculate mean and median coverage of each CpG site
cov <- All_MGMT_Island %>% group_by(SampleID) %>% summarise(AVG_cov = mean(Valid_cov), MED_cov = median(Valid_cov))

MGMT_RunSum <- inner_join(Samples,cov)
```


```{r MGMT_stats, echo=FALSE}


# Summarize and count tumors
Table <- MGMT_RunSum %>%
  dplyr::select(SampleID,Series,Diagnosis) %>%
  distinct() %>%
  dplyr::group_by(Series,Diagnosis) %>%
  tally() %>%
  spread(Diagnosis,n) %>%
  replace(is.na(.), 0) %>%
  adorn_totals(c("row", "col"))

#transpose and name columns
Table <- row_to_names(t(Table),1)

```

Table has been reformated for Latex
```{r Sample_overview, echo=FALSE}

knitr::kable(Table,align = "lccccccc", format="pipe", booktabs=TRUE, caption = "Overview of samples included in this study")

#knitr::kable(Table,align = "lccccccc", format="html", booktabs=TRUE, caption = "Overview of samples included in this study")

#knitr::kable(Table,align = "lcccc", format="latex", caption = "Overview of samples included in this study") %>% kable_styling(latex_options="scale_down")

```

### Overview of samples included in study
```{r, Methylation_overview, echo=FALSE,  eval= FALSE, out.width="100%", fig.cap="Distribution of methylated versus unmethylated samples according to known status prior to nanopore sequencing"}
ggplot(MGMT_RunSum, aes(y=Diagnosis, fill=Known_status))+
  geom_bar(stat = "count")+
  scale_fill_brewer(name="Mehtylation Status", palette="Set1")+
  theme_bw()+
  theme(legend.position = "top")

```

Also simplified version (Figure 1b)

```{r Methylation_overview_simplified, echo=FALSE, eval= TRUE,out.width="100%", fig.cap="Distribution of methylated versus unmethylated samples according to known status prior to nanopore sequencing"}
barplot_data <- MGMT_RunSum %>% dplyr::select(c(SampleID, Diagnosis, Known_status)) %>%
  distinct() %>%
  group_by(Diagnosis, Known_status) %>%
  tally() %>%
  spread(Diagnosis,n) %>%
  replace(is.na(.), 0) %>%
  dplyr::mutate(Other = rowSums(dplyr::select(.,c(LGG_PA,ATRT_SHH, Ganglioglioma, HPC, Medulloblastoma, PCNSL, PXA, Metastasis, Meningioma, Unknown)))) %>%
  dplyr::mutate('IDH glioma' = rowSums(dplyr::select(.,c(Astrocytoma,`Astrocytoma HG`, Oligodendroglioma)))) %>%
  dplyr::select(c(Known_status,'IDH glioma', Glioblastoma, Other)) %>%
  gather(key=Diagnosis, value = Samples, -Known_status)


barplot_data$Diagnosis <- factor(barplot_data$Diagnosis, levels = c("Glioblastoma", 'IDH glioma', "Other"))


ggplot(barplot_data, aes(x=Diagnosis, y= Samples, fill=Known_status, label = Samples))+
  geom_bar(stat = "identity")+
 # coord_flip()+
  scale_fill_brewer(name="Known status", palette="Set1")+
  scale_x_discrete(limits = c("Glioblastoma", "IDH glioma", "Other"))+
  geom_text(size = 5, 
            position = position_stack(vjust = 0.5),
            )+
  theme_classic(base_size = 16)+
  xlab("")+
  theme(legend.position = "top", aspect.ratio = 1 )
```


## Data acquisition. Figure 1c

```{r Coverage_Method, echo=FALSE, fig.cap="Sequencing depth of samples by method of acquisition"}
# Distribution of hits in RIO by Methylation status and sequencing method
ggplot(MGMT_RunSum, aes(x=Method, y=MED_cov))+
  geom_boxplot(aes(fill=Known_status), alpha = 0.2)+
  geom_point(size = 2, position = position_jitterdodge(0.2,0), alpha = 0.7, aes(color=Known_status))+
  #coord_flip()+
  scale_color_brewer(name="Known status", palette = "Set1")+
  scale_fill_brewer(name="Known status", palette="Set1")+
  scale_x_discrete(
    labels = c("Single\nnCATs", "Barcoded\nnCATs","Adaptive\nSampling", "WGSeq"), 
    limits=c("Single", "Barcoded", "AdaptiveSampling","WG_Seq")
    )+
  scale_y_continuous(trans = 'log10', breaks = c(1,10,100,1000), limits = c(0.9,1000))+
  xlab ("")+
  ylab("Median methylation coverage") +
  theme_bw(base_size = 16)+
  theme(legend.position = "top", aspect.ratio = 1 )

```

```{r Coverage_Dataset, echo=FALSE, eval=FALSE, fig.cap="Sequencing depth of methods, group median represented by red diamond. Single sample runs generally have higher sequence depth than barcoded samples or adaptive sampling."}
ggplot(MGMT_RunSum, aes(x=Series, y=MED_cov))+
  geom_boxplot(aes(color=Method), alpha=0.1) +
  geom_point(size = 2, position = position_jitterdodge(0.2,0), alpha = 0.6, aes(color=Method)) +
  stat_summary(fun=median, geom="point", shape="-", size=12, color="black",alpha = 0.8,position = position_dodge(0.8), aes(fill=Method))+
  scale_color_viridis_d(name="Method")+
  scale_fill_discrete(guide="none")+
  labs(fill = "", color = "Method")+
  scale_color_viridis_d()+
  scale_y_log10()+
  xlab ("")+
  ylab("Median methylation depth") +
  theme_bw(base_size = 14) + theme(axis.text.x = element_text(angle = 45, hjust=1))
```

## Direct comparison of Nanopore sequencing and Pyrosequencing

Comparison of individual CpGs as measured by nanopore and pyrosequencing (Figure 2a)

```{r Compare_Individual, echo=FALSE, out.width="100%", fig.cap="Comparison of individual CpGs within exon 1" }
library(ggpubr)

# Extract the CpG sites that correspond to the MGMT Pyro kit sites
All_MGMT_Pyro <- All_MGMT %>% filter(between(Pos, 129467253, 129467272)) %>% select(1:4)

# Add the Pyro kit results to the table
FOUR_CpGs <- inner_join(Pyro, All_MGMT_Pyro)
FOUR_CpGs$Pos <- as.factor(FOUR_CpGs$Pos)
FOUR_CpGs$CpG <-  ifelse(FOUR_CpGs$Pos == 129467254, "CpG 76",
                                      ifelse(FOUR_CpGs$Pos == 129467257, "CpG 77",
                                             ifelse(FOUR_CpGs$Pos == 129467261, "CpG 78", "CpG 79")))

# Filter out sites with coverage below 3
FOUR_CpGs <- FOUR_CpGs %>% filter(Valid_cov > 2)

# plot results
ggplot(FOUR_CpGs, aes(x=Percent_Meth_Pyro, y=Methylation_percent, color = CpG))+
  geom_smooth(method = lm, color="black")+
  stat_regline_equation(label.y = 20, label.x = 60,aes(label = ..rr.label..),color = "black")+
 # stat_cor(method = "pearson",label.y = 10, label.x = 40, color = "black")+
  geom_point(aes(size=Valid_cov), alpha = 0.6)+
  scale_size_continuous(name="Coverage",breaks = c(5,15,25,100,150), range = c(1,10))+
  xlab("Pyro sequencing (% methylated)")+
  ylab("Nanopore sequencing (% methylated)")+
  xlim(c(0,100))+
  ylim(c(0,100))+
  theme_bw(base_size = 16)+
  scale_color_brewer(palette = "Set2", guide="none")+
  facet_wrap(.~CpG)+
  theme(aspect.ratio = 1)

```

Group average as measured by nanopore and pyrosequencing (Figure 2b)
```{r Compare_Group, echo=FALSE, out.width="100%", fig.width=10, fig.height=5, fig.cap="Comparison of nanopore sequencing and Pyrosequencing results of 4 CpGs in exon 1 of the MGMT promoter. Plotted values are average methylation of 4 CpGs. Black horizontal and vertical lines mark the 9 percent cut-off value between methylated and unmethylated samples, as determined by pyrosequencing."}
library(ggpubr)

#Average the 4 CpG sites and filter samples with average coverage below 3)
FOUR_CpGs_Group <- FOUR_CpGs %>% 
                      group_by(SampleID) %>% 
                        summarise(Average_Nano=mean(Methylation_percent), 
                        Average_Pyro=mean(Percent_Meth_Pyro),
                        Average_Cov=mean(Valid_cov)) %>%
                        inner_join(Samples) %>%
                    filter(Average_Cov>2) 
                    
# plot the data
ggplot(FOUR_CpGs_Group, aes(x=Average_Pyro, y=Average_Nano)) +
  geom_smooth(method = "lm", color="black") +
  geom_point(aes(
    size=Average_Cov
    ,color=Known_status
    ), 
    alpha = 0.6 )+
  stat_regline_equation(label.y = 15, label.x = 50,aes(label = ..rr.label..),color = "black", size=5)+
 # stat_cor(method = "pearson",label.y = 10, label.x = 50)+
  scale_color_brewer(palette = "Set1", guide="none", name="Known Status")+
  scale_size_continuous(name="Coverage",breaks = c(5,10,50,100,150), range = c(1,10) )+
  guides(colour = guide_legend(override.aes = list(size=6)), position = "top")+
  xlab("Pyro sequencing (% methylated)")+
  ylab("Nanopore sequencing (% methylated)")+
 # geom_hline(yintercept = 20)+
  geom_vline(xintercept = 10)+
  theme_bw()+ 
  coord_fixed()


```


Make logistic regression model to predict Pyro results from nanopore results. Plot ROC curve (Figure 2c)

```{r Logistic_regression, echo=FALSE}
### Use the Radium samples to make a training model 

#make training data
Train <- FOUR_CpGs_Group %>% select(Average_Nano,Known_status)
Train <- Train[order(Train$Average_Nano),]
Train$P <- ifelse(Train$Known_status=="Methylated",1,0)

#make test data
test <- All_MGMT_Pyro %>% 
  group_by(SampleID) %>% 
  summarise(Average_Nano=mean(Methylation_percent), 
            Average_Cov=mean(Valid_cov)) %>%
  inner_join(Samples) %>%
  filter(Average_Cov>2) %>%
  filter(Series != "Radium") %>%
  select(Average_Nano,Known_status)

test$P <- ifelse(test$Known_status=="Methylated",1,0)
test <- test[order(test$Average_Nano),]

#make the training model

model <- glm(P ~ Average_Nano, family = "binomial", data = Train)

summary(model)

#predict the other samples
predicted <- predict(model, test, type = "response")

# plot the training data en fitted curve from model

ggplot(Train, aes(x=Average_Nano, y=P))+
  geom_line(aes(x=Average_Nano, y=model$fitted.values))+
  geom_point(aes(color=Known_status),size=3, alpha=0.7)+
  geom_vline(xintercept = 23)+
  scale_color_brewer(name="Known Status", palette = "Set1")+
  ylab("")+
  xlab("Nanopore sequencing\n(% methylated)")+
  theme_bw(base_size = 16)

#### make the ROC cur
library(pROC)

ROC.Train <- roc(Train$P, 
    model$fitted.values, 
    legacy.axes = T, 
    percent = T, 
    print.auc = T)

ROC.Predict <- roc(test$P, 
                   predicted,
                   legacy.axes = T, 
                   percent = T, 
                   print.auc = T)

rocs_Pyro <- list()

rocs_Pyro[["Retrospective nCATs"]] <- ROC.Train
rocs_Pyro[["Other"]] <- ROC.Predict

##### MAKE ROC curves

#ggroc(ROC.Train) # training set
#ggroc(ROC.Predict) # prediction set

auc(rocs_Pyro$`Retrospective nCATs`)
auc(rocs_Pyro$Other)

ggroc(rocs_Pyro, aes = c("colour","linetype"),size=1.5, alpha=0.8)+
  scale_color_manual("",
                     values = c("black", "darkgrey"),
                     labels=c("Retrospective nCATs\nAUC = 99.15 %",
                              "\nOther\nAUC = 95.08 %"))+
  scale_linetype_manual("", 
                        values=c("solid","dashed"), 
                        labels=c("Retrospective nCATs\nAUC = 99.15 %",
                              "\nOther\nAUC = 95.08 %"))+
  theme_bw(base_size = 16)+
  theme(legend.position = c(.95, .65),
        legend.justification = c("right", "top"),
        legend.margin = margin(6, 6, 6, 6))
```

Plot results of logistic recression classifier (Figure 2d)

```{r BeeSwarm, echo=FALSE, out.width="100%",fig.cap="Average nanopore sequencing percentage of CpGs 76-79 agains the Pyrosequencing classification"}

library(ggbeeswarm)

# Add the Pyro kit results to the table
FOUR_CpGs_Beeswarm <- All_MGMT %>% filter(between(Pos, 129467253, 129467272)) %>% select(1:4)

FOUR_CpGs_Beeswarm <- All_MGMT %>% filter(between(Pos, 129467253, 129467272)) %>% 
  select(1:4) %>%
  group_by(SampleID) %>% 
  summarise(Average_Nano=mean(Methylation_percent), 
            Average_Cov=mean(Valid_cov)) %>%
  left_join(Samples) %>%
  filter(Average_Cov>2) 

FOUR_CpGs_Beeswarm$Series <-  ifelse(FOUR_CpGs_Beeswarm$Series == "Radium", "Retrospective nCATs", "Other")

#FOUR_CpGs_Beeswarm <- FOUR_CpGs_Beeswarm %>% filter(Series == "Retrospective nCATs")
                         
ggplot(FOUR_CpGs_Beeswarm, aes(x=Known_status, y=Average_Nano, color = Known_status))+
  geom_rect(xmin=0, xmax = 2.9, ymin=10, ymax=30, alpha=.2, fill='light grey', color = "black")+
  geom_hline(yintercept = 20, linetype = 2)+
  geom_beeswarm(
  #  aes(size = Average_Cov),
    alpha = 0.6, cex=3)+
  #scale_size_continuous(name="Coverage",breaks = c(5,10,50,100), range = c(1,10) )+
  scale_color_brewer(palette = "Set1", guide="none")+
  xlab("Known methylation status")+
  ylab("NP average methylation %\nCpGs 76-79")+
  facet_grid(~factor(Series, levels = c("Retrospective nCATs", "Other")))+
  theme_bw()+
  theme(aspect.ratio = 1)
```

Examine methylation on STP27 sites (Figure 3a)
```{r STP27, echo=FALSE, out.width="100%"}
#STP27_CpGs <- read.csv("MGMT_R/R-markdown/Data/STP27_All_samples.csv")


# Make STP27 table

STP27 <- All_MGMT %>% 
  filter(Pos == 129466944 | Pos == 129467310) %>% 
  dplyr::select(1:2,4) %>% 
  pivot_wider(names_from = Pos, values_from=Methylation_percent) %>%
  inner_join(MGMT_RunSum) %>%
  filter(AVG_cov>2) %>%
  na.omit()

colnames(STP27) <- c("SampleID", "CpG_31", "CpG_84", "Series", "Method", "Known_status", "Diagnosis","AVG_cov","MED_cov" )

STP27$Series <- ifelse(STP27$Series == "Rapid-CNS", "Rapid-CNS", "Other")

ggplot(STP27, aes(y=CpG_31, x=CpG_84, color = Known_status)) +
  #  stat_function(fun = fun.1, color = "grey", size = 2) +
  geom_point(size = 3, alpha = 0.6)+
  stat_ellipse(aes(fill = Known_status), level = 0.95, geom = "polygon", alpha = 0.1)+
  scale_fill_discrete(guide = "none")+
 # stat_ellipse(level = 0.50, na.rm=T, linetype = 2)+
  #  scale_size_continuous(name="Depth",breaks = c(5,15,30), range = c(1,8))+
  scale_color_brewer(name="Known Status", palette = "Set1")+
  scale_x_continuous(breaks=c(0,20,40,60,80,100))+
  scale_y_continuous(breaks=c(0,20,40,60,80,100))+
  xlab("CpG 84 (cg12981137) % methylated (NP)")+
  ylab("CpG 31 (cg12434587)\n% methylated (NP)")+
  guides(colour = guide_legend(override.aes = list(size=5)))+
  theme_bw(base_size = 16)+
  facet_wrap(~factor(Series, levels = c("Rapid-CNS", "Other")))+
  theme(aspect.ratio = 1, 
        panel.grid.minor.x = element_blank(),
        panel.grid.minor.y = element_blank())

#### Average methylation over STP27 sites


STP27 <- STP27 %>% mutate(STPMean = rowMeans(select(., starts_with("CpG")), na.rm = TRUE))                                      




```

Make logistic regression model based on STP27 sites. Plot ROC curve (Figure 3b)
```{r STP27 regression model, echo=FALSE, out.width="100%"}
###################### STP27 regression model
STP27$P <- ifelse(STP27$Known_status=="Methylated",1,0)

Stp27.train <- STP27 %>% filter(Series == "Rapid-CNS") 
Stp27.test <- STP27 %>% filter(Series != "Rapid-CNS") 


stp.model <- glm(P ~ CpG_31 + CpG_84, family = "binomial", data = Stp27.train)

stp.predicted <- predict(stp.model, Stp27.test, type = "response")



##### Stp ROC curves

#### make the ROC cur
library(pROC)

stp.ROC.Train <- roc(Stp27.train$P, 
                     stp.model$fitted.values, 
                     legacy.axes = T, 
                     percent = T, 
                     print.auc = T)

stp.ROC.Predict <- roc(Stp27.test$P, 
                    stp.predicted,
                   legacy.axes = T, 
                   percent = T, 
                   print.auc = T)

auc(stp.ROC.Train)
auc(stp.ROC.Predict)

rocs.STP <- list()

rocs.STP[["Rapid-CNS"]] <- stp.ROC.Train
rocs.STP[["Other"]] <- stp.ROC.Predict

ggroc(rocs.STP, aes = c("colour","linetype"),size=1.5, alpha = 0.7)+
  scale_color_manual("",
                     values = c("black", "darkgrey"),
                     labels=c("Rapid-CNS\nAUC = 97.2 %",
                                 "\nOther\nAUC = 94.4 %"))+
  scale_linetype_manual("", 
                        values=c("solid","dashed"), 
                        labels=c("Rapid-CNS\nAUC = 97.2 %",
                                 "\nOther\nAUC = 94.4 %"))+
  theme_bw(base_size = 16)+
  theme(legend.position = c(.95, .65),
        legend.justification = c("right", "top"),
        legend.margin = margin(6, 6, 6, 6))
```

Compare results of STP27 logistic regression model (Figure 3c)
```{r STP predictions, echo=FALSE, out.width="100%",fig.cap=""}

STP27$predictions <- predict(stp.model, STP27)

ggplot(STP27, aes(x = Known_status, y = predictions, color = Known_status))+
  #geom_rect(xmin=0, xmax = 2.9, ymin=10, ymax=30, alpha=.2, fill='light grey', color = "black")+
  geom_hline(yintercept = 0, linetype = 2)+
  geom_beeswarm(
    #  aes(size = Average_Cov),
    alpha = 0.6, cex=3, size=3)+
  #scale_size_continuous(name="Coverage",breaks = c(5,10,50,100), range = c(1,10) )+
  scale_color_brewer(palette = "Set1", guide="none")+
  xlab("Known methylation status")+
  ylab("Linear prediction \n CpG31 and CpG84")+
  facet_grid(~factor(Series, levels = c("Rapid-CNS", "Other")))+
  theme_bw(base_size = 16)+
  theme(aspect.ratio = 1)

```



## Correlation plot of CpG sites in MGMT CpG island. (Figure omitted)

```{r Corrolation_plot, echo=FALSE, eval=TRUE}
library(corrplot)
# Extract the CpGs in the CpG island (98)
All_MGMT_island <- All_MGMT %>% filter(between(Pos, 129466683, 129467448))

MGMT_long_island <- All_MGMT_island %>% 
  select(c("SampleID", "Pos", "Methylation_percent")) %>%
  pivot_wider(names_from = Pos, values_from=Methylation_percent) %>%
  column_to_rownames("SampleID")

colnames(MGMT_long_island ) <- c(1:98)

MGMT_long_island <- na.omit(MGMT_long_island)
# correlation of CpGs
corrplot(cor(MGMT_long_island),
         method = "color",
         order = "original",
         #hclust.method = "ward.D",
         tl.cex = 0.4,
         tl.col = 'black')
```

Heatmap of all samples (Figure 4c)
```{r Heatmap, echo=FALSE,  out.width="100%", out.height="100%", fig.cap="Clustered heatmap of all samples based on nanopore sequencing of CpG island of the MGMT promoter"}
library(stringi)
library(pheatmap)
library(grid)
library(RColorBrewer)
library(tibble)

MGMT_long_island <- All_MGMT_Island %>%
  dplyr::select(c("SampleID", "Pos", "Methylation_percent")) %>%
  pivot_wider(names_from = Pos, values_from=Methylation_percent) %>%
  column_to_rownames("SampleID")

colnames(MGMT_long_island ) <- c(1:98)




# 
ann_colors <- list(
  "Known status" = c("Unmethylated"="#377EB8", 
                   "Methylated"="#E41A1C"), 
  Diagnosis = c("Glioblastoma"="#31688EFF", 
                "IDHglioma"="#440154FF", 
                "Other"="#35B779FF"),
  "Assay CpG sites"= c("STP-27" = "black", 
             "MGMT-Pyro" = "darkgrey", 
             " "  = "white"),
  "Original evaluation" = c("STP-27" = "#addc30",
                            "MGMT-Pyro" = "#21918c")
)

col_anno <- data.frame(Method=c(c(1:30), 
                                "STP-27",c(32:75), 
                                "MGMT-Pyro",
                                "MGMT-Pyro",
                                "MGMT-Pyro",
                                "MGMT-Pyro", 
                                c(80:83),
                                "STP-27",
                                c(85:98)
                                ))
col_anno$Method <- ifelse(col_anno$Method == "STP-27","STP-27", ifelse(col_anno$Method == "MGMT-Pyro", "MGMT-Pyro", " "))

names(col_anno) <- "Assay CpG sites"

#################### New row anno to include detection method 

row_anno <- MGMT_RunSum %>%
  dplyr::select(SampleID,Known_status,Series,Diagnosis) %>%
  distinct() %>% column_to_rownames("SampleID") 
# %>%
#   filter(Diagnosis == "Glioblastoma")

row_anno <- na.omit(row_anno[rownames(MGMT_long_island),])
MGMT_long_island <- MGMT_long_island[rownames(row_anno),]

row_anno$Diagnosis <- ifelse(
  row_anno$Diagnosis == "Astrocytoma" | 
    row_anno$Diagnosis == "Oligodendroglioma" | 
    row_anno$Diagnosis == "Astrocytoma HG", "IDHglioma", 
  ifelse(row_anno$Diagnosis == "Glioblastoma", "Glioblastoma", "Other"))

row_anno$Series <- ifelse(row_anno$Series == "Rapid-CNS", "STP-27", "MGMT-Pyro")

colnames(row_anno) <- c("Known status", "Original evaluation", "Diagnosis")

pheatmap(MGMT_long_island,
         cluster_rows = T,
         cluster_cols = F,
         clustering_method = "ward.D",
         cutree_rows = 4,
         treeheight_row = 30,
         border_color = NA,
         scale = "none",
         drop_levels = T,
         color = rev(brewer.pal(n = 10, name = "Spectral")),
         fontsize_col = 8,
         fontsize_row = 4,
         fontsize = 12,
         legend = T,
         show_rownames = F,
         legend_breaks = seq(0,100,10),
         annotation_col = col_anno,
         annotation_row = row_anno,
         annotation_colors = ann_colors,
         legend_labels = c("0%","10%", "20%", "30%", "40%", "50%", "60%", "70%", "80%", "90%", "100%")
)


```

Per site methylation values in methylated versus unmethylated samples and p-values (Figure 4a and 4b)

```{r Dotline, echo=FALSE,  out.width="100%" }
# Dotplot comparing methylation on every CpG site for methylated and unmethylated samples

library(purrr)
library(rstatix)
library(stringr)

########################### Plot mean methylation per cpg and standard deviation

x <- All_MGMT_Island %>% 
     select(c("SampleID", "Pos", "Methylation_percent")) %>%
      pivot_wider(names_from = Pos, values_from=Methylation_percent) 

y2 <- MGMT_RunSum %>% dplyr::select(SampleID,Known_status)

x2 <- left_join(x, y2) %>% 
     filter(!str_detect(SampleID, "^T")) %>% # Filter out DenStem and WGS samples as they are different biopsies analysed for pyro and nano
  na.omit() 

dotplot <- x2  %>% 
  gather(key = "CpG", value = "percentMeth", -c(Known_status,SampleID)) %>% 
  group_by(Known_status, CpG) %>% 
  summarise(Average_Methylation = mean(percentMeth), SD_Methylation = sd(percentMeth))


dotplot$CpG_row <- as.integer(rep(c(1:98),2))

regions <- tibble(x1 = 75.5, x2 = 79.5, y1 = -10, y2 = 99, y3 =90, x3 = 31, y4 = 90, x4 = 84)

ggplot(dotplot, aes(x=CpG_row, y=Average_Methylation, group=Known_status))+
  geom_rect(data = regions,   # make a transparent box around the MGMT Pyro sites
            inherit.aes = FALSE,
            mapping = aes(xmin = x1, xmax = x2,
                          ymin = y1, ymax = y2),
            color = "transparent",
            fill = "black",
            alpha = .2)+
  geom_text(mapping = aes(y=75,x=31), label = "*", size = 10)+ # mark the two STP27 sites 
  geom_text(mapping = aes(y=80,x=84), label = "*", size = 10)+
  geom_point(aes(color=Known_status), position = position_dodge(width = 0.9), size = 2)+
  geom_linerange(aes(ymin=Average_Methylation-SD_Methylation, ymax=Average_Methylation+SD_Methylation,color=Known_status),position = position_dodge(width = 0.9))+
  scale_color_brewer(palette = "Set1", name = "Known status")+
  ylab("Average methylation (%)")+
  xlab("CpG in MGMT promoter island")+
  geom_line(aes(color=Known_status))+
  scale_x_continuous(limits = c(0,99),breaks = c(1,10,20,30,40,50,60,70,80,90,98))+
    #geom_smooth(aes(color=Known_status), show.legend = FALSE)+
  theme_bw(base_size = 12)+
  theme(legend.position="top", aspect.ratio = 0.15)

######################################################
### t-test for every CpG

CpGs <- as.factor(x3$CpG)

P_list <- lapply(x2[,CpGs], function(x) t.test(x ~ x2$Known_status, var.equal = FALSE)$p.value)
P_frame <- as.data.frame(unlist(P_list))
names(P_frame) <- "p.val"

P_frame$adj.p.val <- p.adjust(P_frame$p.val, method = "bonferroni", n = length(P_frame$p.val))

P_frame <- P_frame[1:98,]


P_frame$CpG <- as.integer(1:98)

regions <- tibble(x1 = 75.5, x2 = 79.5, y1 = 1e-25, y2 = 5, y3 =0.00001, x3 = 31, y4 = 0.000000001, x4 = 84)


ggplot(P_frame, aes(x=CpG, y=adj.p.val))+
  geom_point(size = 4)+
  geom_rect(data = regions,   # make a transparent box around the MGMT Pyro sites
            inherit.aes = FALSE,
            mapping = aes(xmin = x1, xmax = x2,
                          ymin = y1, ymax = y2),
            color = "transparent",
            fill = "black",
            alpha = .2)+
  geom_text(data = regions, mapping = aes(y=y3,x=x3), label = "*", size = 10)+ # mark the two STP27 sites
  geom_text(data = regions, mapping = aes(y=y4,x=x4), label = "*", size = 10)+
  geom_line()+
  ylab('Adjusted p-value')+
  xlab("CpG in MGMT promoter island")+
  scale_y_continuous(trans = 'log10')+
  scale_x_continuous(breaks = c(1,10,20,30,40,50,60,70,80,90,98))+
  geom_hline(yintercept = 0.01)+  
  theme_bw(base_size = 12)+
  theme(aspect.ratio = 0.15)

```


Heatmap of only GBM samples (Figure 5a)

```{r HeatmapGBM, echo=FALSE,  out.width="100%", out.height="100%", fig.cap="Heatmap showing unsupervised clustering of glioblastoma samples based on nanopore sequencing of the CpG island in the MGMT promoter"}

GBMs <- MGMT_RunSum %>% dplyr::select(SampleID,Diagnosis,Series) %>% dplyr::filter(Diagnosis == "Glioblastoma" ) %>% distinct()


MGMT_long_island_GBM <- MGMT_long_island[GBMs$SampleID,]



row_anno_GBM <- row_anno[1]

 pheatmap(MGMT_long_island_GBM,
          cluster_rows = T,
         cluster_cols = F,
         clustering_method = "ward.D",
         cutree_rows = 2,
         treeheight_row = 20,
         border_color = NA,
         scale = "none",
         drop_levels = T,
         color = rev(brewer.pal(n = 10, name = "Spectral")),
         fontsize_col = 8,
         fontsize_row = 4,
         fontsize = 12,
         legend = T,
         show_rownames = F,
         legend_breaks = seq(0,100,10),
         annotation_col = col_anno,
          annotation_row = row_anno_GBM,
         annotation_colors = ann_colors,
         legend_labels = c("0%","10%", "20%", "30%", "40%", "50%", "60%", "70%", "80%", "90%", "100%")
         
)



```

K-means clustering (omitted)

```{r Kmeans, echo=FALSE,  out.width="100%", out.height="100%", fig.cap="K-means clustering of glioblastoma samples"}
#GBMs <- MGMT_RunSum %>% select(SampleID,Diagnosis) %>% filter(Diagnosis == "Glioblastoma") %>% distinct()

#Metsum_ROI_long_GBM <- na.omit(Metsum_ROI_long[GBMs$SampleID,])


pheatmap(na.omit(MGMT_long_island_GBM),
         cluster_rows = T,
         cluster_cols = F,
         clustering_method = "ward.D",
         cutree_rows = 3,
         scale = "none",
         border_color = NA,
         color = rev(brewer.pal(n = 10, name = "Spectral")),
         fontsize = 10,
         fontsize_col = 8,
         fontsize_row = 4,
          kmeans_k = 3,
         show_rownames = F,
         treeheight_row = 40,
         legend = T,
         legend_breaks = seq(0,100,10),
        # annotation_col = col_anno,
         annotation_colors = ann_colors,
         legend_labels = c("0%","10%", "20%", "30%", "40%", "50%", "60%", "70%", "80%", "90%", "100%")
)
```
## Survival analysis

Kaplan Meier survival analysis (Figures 5b and 5c)

```{r KaplanMeier, echo=FALSE,  out.width="100%", out.height="100%", fig.cap="Kaplan-Meier Overall Survival curves"}
#setwd("/home/vml/Documents")

# perform sample clustering of the GBM samples, save the sample grouping
res <- pheatmap(MGMT_long_island_GBM,
         cluster_rows = T,
         cluster_cols = F,
         clustering_method = "ward.D",
         cutree_rows = 2,
         scale = "none",
         drop_levels = T,
         )

MGMT_clusters <- data.frame(cluster=cutree(res$tree_row, k=2))

MGMT_clusters$SampleID <- rownames(MGMT_clusters)

cluster_surv <- inner_join(MGMT_clusters, Patient_meta, by="SampleID") # add cluster info to patient metadata

cluster_surv$months_to_event <- as.numeric(cluster_surv$months_to_event)

cluster_surv_filt <- cluster_surv %>% filter(Diagnosis == "GBM" & # only include GBM patients
                                               Age < 76 & # Remove the patients over 75 years of age
                                               Reopr == 0 & # Only include primary cases
                                               Onc.treatment == "Stupp") # Only include patients that recieve full treatment
                                               
cluster_surv_filt$cluster <- ifelse(cluster_surv_filt$cluster == 1,2,1) # workaraound to fix group ordering in KM-plot

####################### Survival
library(survival)
library(survminer)
library(lubridate)

# Reopr filtered out
fit1_filt <- survfit( Surv(cluster_surv_filt$months_to_event, cluster_surv_filt$censor) ~ cluster_surv_filt$Pyro_state)
fit2_filt <- survfit( Surv(cluster_surv_filt$months_to_event, cluster_surv_filt$censor) ~ cluster_surv_filt$cluster)

# fit3_filt <- survfit( Surv(cluster_surv_filt$months_to_event, cluster_surv_filt$censor) ~ cluster_surv_filt$Resection)




# Kaplan-Meier plot
## Filtered
# By pyrosequencing results
ggsurvplot(
  fit = fit1_filt,
  data = cluster_surv_filt,
  palette = c("darkorange3","cornflowerblue"),
  xlab = "OS, Months",
  ylab = "Overall survival probability",
  legend.title = "Pyrosequencing",
  legend.labs = c("Methylated", "Unmethylated"),
  pval = TRUE,
  pval.coord = c(0, 0.1),
  font.x = c(14, "bold", "black"),
  font.y = c(14, "bold", "black"),
  font.legend = 16,
  surv.median.line = "hv",
  risk.table = TRUE)

# By nanopore clustering
ggsurvplot(
  fit = fit2_filt,
  data = cluster_surv_filt,
  palette = c("darkorange3","cornflowerblue"),
  xlab = "OS, Months",
  ylab = "Overall survival probability",
  legend.title = "Nanopore",
  legend.labs = c("Cluster 2", "Cluster 1"),
  pval = TRUE,
  pval.coord = c(0, 0.1),
  font.x = c(14, "bold", "black"),
  font.y = c(14, "bold", "black"),
  font.legend = 16,
  surv.median.line = "hv",
  risk.table = TRUE)

# # By resection type
# ggsurvplot(
#   fit = fit3_filt,
#   data = cluster_surv_filt,
#   
#   #palette = c("cornflowerblue","darkorange3"),
#   xlab = "OS, Months",
#   ylab = "Overall survival probability",
#   legend.title = "Nanopore",
#   #legend.labs = c("Cluster 1", "Cluster 2", "Cluster 3"),
#   #legend.labs = c("Cluster 1", "Cluster 2"),
#   pval = TRUE,
#   pval.coord = c(0, 0.1),
#   font.x = c(14, "bold", "black"),
#   font.y = c(14, "bold", "black"),
#   risk.table = TRUE)

```

